src/buildercore/project/files.py

```diff
@@ -42,6 +42,8 @@ def write_project_file(new_project_data, project_file):
 @testme
 def all_projects(project_file): #, project_file=config.PROJECT_FILE):
     allp = utils.ordered_load(open(project_file))
+    assert allp is not None, ("Project file %s seems to be empty" % project_file)
```

Comment:
I would suggest if there is an empty file to just return empty values: `return ({}, [])`

however, if there is a non-empty project file *without* a defaults section, then die.